deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobIT.java

```diff
@@ -391,6 +392,35 @@ public void ingestDeepHierarchy() {
 
 		assertClickCount(nestingDepth + 1);
 	}
+	
+	@Test (expected = TransactionCancelledException.class)
+	public void transactionCancelledTest() {
+		String label = "testwork";
+
+		// Construct the deposit model with work object
+		Model model = job.getWritableModel();
+		Bag depBag = model.createBag(depositPid.getRepositoryPath());
+
+		// Constructing the folder in the deposit model with a title
+		PID workPid = repository.mintContentPid();
+		Bag workBag = model.createBag(workPid.getRepositoryPath());
+		
+		// Add incorrect resource type to bag, which causes a DepositException to be thrown
+		// and the tx to be cancelled
+		workBag.addProperty(RDF.type, Cdr.FileObject);
+		workBag.addProperty(CdrDeposit.label, label);
+
+		PID mainPid = addFileObject(workBag, FILE1_LOC, FILE1_MIMETYPE, FILE1_SHA1);
+
+		depBag.add(workBag);
+
+		workBag.asResource().addProperty(Cdr.primaryObject,
+				model.getResource(mainPid.getRepositoryPath()));
+
+		job.closeModel();
+
+		job.run();
```

Comment:
One other suggestion, it would be a bit reassuring to make sure that the transaction has ended and that the work does not exist in fedora after job.run() has completed, just to ensure it did the work of cancelling the tx